src/waveform/widgets/allshader/waveformwidget.cpp

```diff
@@ -21,6 +21,11 @@
 
 namespace allshader {
 
+std::unique_ptr<rendergraph::BaseNode> convert(
+        std::unique_ptr<WaveformRendererSignalBase>&& pRenderer) {
+    return std::unique_ptr<rendergraph::BaseNode>(pRenderer.release()->asNode());
+}
```

Comment:
I'm split on this solution. We can either do this, which involves a virtual function call, or we can do a `qobject_cast` or `dynamic_cast`. Do you have an estimate on how it compares? Is it possible fix the object hierarchy instead so we don't have to do the horizontal cast?